libfreerdp/core/nla.c

```diff
@@ -636,32 +636,21 @@ static BOOL nla_adjust_settings_from_smartcard(rdpNla* nla)
 	}
 
 setup_pin:
-	if (!settings->SmartcardPin)
+	if (!settings->Password)
 	{
-		if (settings->SmartcardEmulation)
-		{
-			if (!freerdp_settings_set_string(settings, FreeRDP_SmartcardPin, ""))
-			{
-				WLog_ERR(TAG, "error setting pin");
-				return FALSE;
-			}
-		}
-		else
-		{
 			freerdp* instance = nla->instance;
-			if (!instance->AuthenticateEx)
-			{
-				WLog_ERR(TAG, "no pin configured and no instance->AuthenticateEx callback");
-				return TRUE;
-			}
-
-			if (!instance->AuthenticateEx(instance, &settings->Username, &settings->SmartcardPin,
-			                              &settings->Domain, AUTH_SMARTCARD_PIN))
-			{
-				WLog_ERR(TAG, "no pin code entered");
-				return TRUE;
-			}
-		}
+		    auth_status reason = utils_authenticate(instance, AUTH_NLA, TRUE);
```

Comment:
Shouldn't this be `AUTH_SMARTCARD_PIN`?

Reply:
@hardening might shed some light on this. Is `user` and `domain` required for `smartcard-nla` then no, if not then yes.

Reply:
That should definitely be `AUTH_SMARTCARD_PIN` as the pin is required to logon as the server do some checks once the NLA logon has passed using the smartcard channel.

Reply:
@hardening right. I´ve checked, the `user` and `domain` field should already be prefilled from the certificate, right?